cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandlerInternal.java

```diff
@@ -107,14 +107,24 @@ public void getArtifactBytes(HttpRequest request, HttpResponder responder,
                                @PathParam("namespace-id") String namespaceId,
                                @PathParam("artifact-name") String artifactName,
                                @PathParam("artifact-version") String artifactVersion,
-                               @QueryParam("scope") @DefaultValue("user") String scope) throws Exception {
+                               @QueryParam("scope") @DefaultValue("user") String scope,
+                               @QueryParam("lastModified") @DefaultValue("0") String lastModified) throws Exception {
 
     NamespaceId namespace = validateAndGetScopedNamespace(Ids.namespace(namespaceId), scope);
     ArtifactId artifactId = new ArtifactId(namespace.getNamespace(), artifactName, artifactVersion);
     ArtifactDetail artifactDetail = artifactRepository.getArtifact(Id.Artifact.fromEntityId(artifactId));
     Location location = artifactDetail.getDescriptor().getLocation();
-    responder.sendContent(HttpResponseStatus.OK, new LocationBodyProducer(location),
-                          new DefaultHttpHeaders().add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_OCTET_STREAM));
+
+    io.netty.handler.codec.http.HttpHeaders headers = new DefaultHttpHeaders()
+      .add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_OCTET_STREAM)
```

Comment:
Use the netty `HttpHeaderNames` and `HttpHeaderValues` instead.

Reply:
done